eventstore/model.go

```diff
@@ -317,24 +344,67 @@ func (t *Txn) Discard() {
 	t.discarded = true
 }
 
+type structWithID struct {
+	ID string
+}
+
 func getEntityID(t interface{}) core.EntityID {
-	v := reflect.ValueOf(t)
-	if v.Type().Kind() != reflect.Ptr {
-		v = reflect.New(reflect.TypeOf(v))
-	}
-	v = v.Elem().FieldByName(idFieldName)
-	if !v.IsValid() || v.Type() != reflect.TypeOf(core.EmptyEntityID) {
-		panic("invalid instance: doesn't have EntityID attribute")
+	// JSONMODE: assumes true. make `getEntity` a method of `Model` so maybe can access
+	// the attr, or keep it as a func and receive jsonMode as param
+	jsonMode := true
+	if jsonMode {
+		partialStruct := &structWithID{ID: "YOUDONTHAVEID"}
+		strJson := t.(*string)
+		if err := json.Unmarshal([]byte(*strJson), partialStruct); err != nil {
+			panic(err)
+		}
+		if partialStruct.ID == "YOUDONTHAVEID" {
+			panic("invalid instance: doesn't have an ID attribute")
+		}
+		// Do some extra checking that partialStruct.ID is empty, or a valid uuid, etc.
+
+		return core.EntityID(partialStruct.ID)
+
+	} else {
+
+		v := reflect.ValueOf(t)
+		if v.Type().Kind() != reflect.Ptr {
+			v = reflect.New(reflect.TypeOf(v))
+		}
+		v = v.Elem().FieldByName(idFieldName)
+		if !v.IsValid() || v.Type() != reflect.TypeOf(core.EmptyEntityID) {
+			panic("invalid instance: doesn't have EntityID attribute")
+		}
+
+		return core.EntityID(v.String())
```

Comment:
`getEntityID`, in JSONMode should interpret the json string of the instance to see if has an `ID` attribute, and if not empty if has uuid format. (Some hacks here to distinguish between empty ID from ommited field... just to get the idea)